corehq/motech/repeaters/models.py

```diff
@@ -266,6 +269,10 @@ def __get__(self, obj, objtype=None):
     def __set__(self, obj, value):
         if self.choices and value not in self.choices:
             raise ValueError(f"{value!r} not in {self.choices!r}")
+        if self.schema and not isinstance(value, self.schema):
+            raise TypeError(
+                f"Expected {self.name} to be of type {self.schema.__name__} but got {type(value).__name__}"
+            )
```

Comment:
When I was testing it with Django classes, I figured that Django ORM serialized the object into JSON and saves it. So I thought it is not required. But it would definitely not work anything other than Model classes. So have added the serialization logic https://github.com/dimagi/commcare-hq/pull/30348/commits/37a76f4057d41ffc6287ba8b13a050512afad4ca